### PR TITLE
Refactor find_switch_for_lights and more small refactors

### PR DESCRIPTION
**Pull request by basnijholt**
I would like to merge this before https://github.com/basnijholt/adaptive-lighting/pull/487

### CODE REVIEW

**Comment by basnijholt** [comment]
I just realized this might not work in the following case:
A light is marked as manually controlled, then someone changes the state of the light OUTSIDE of Home Assistant. Then the timer should reset, however, it doesn't do that now.

File: custom_components/adaptive_lighting/switch.py [hunk 1]
```diff
@@ -239,53 +239,69 @@ def _split_service_data(service_data, adapt_brightness, adapt_color):
...
+    hass: HomeAssistant, lights: list[str], is_on: bool
+) -> list[AdaptiveSwitch]:
     config_entries = hass.config_entries.async_entries(DOMAIN)
     data = hass.data[DOMAIN]
-    switches = {}
+    switches = []
     for config in config_entries:
-        # this check is necessary as there seems to always be an extra config
-        # entry that doesn't contain any data. I believe this happens when the
-        # integration exists, but is disabled by the user in HASS.
```
Comment (th3w1zard1) [inline custom_components/adaptive_lighting/switch.py:254]: Could we leave these comments here so it's apparent why we're making the check? Or at least shorten.

File: custom_components/adaptive_lighting/switch.py [hunk 2]
```diff
@@ -239,53 +239,69 @@ def _split_service_data(service_data, adapt_brightness, adapt_color):
...
         if config.entry_id in data:
             switch = data[config.entry_id]["instance"]
+            if is_on and not switch.is_on:
+                continue
```
Comment (th3w1zard1) [inline custom_components/adaptive_lighting/switch.py:252]: Why is this necessary?

**Review by th3w1zard1 — COMMENTED** [review]
It seems odd to have two separate functions defined when all they do is call the logger and raise a valueerror, other than the reviews I left this looks good.

File: custom_components/adaptive_lighting/switch.py (hunk 2 above)
Comment (basnijholt) [reply custom_components/adaptive_lighting/switch.py:252]: To use this function to extract a switch that is on. This might be useful in the case where a light appears in multiple switches, however, currently only one of those switches is on.

Comment (th3w1zard1) [reply custom_components/adaptive_lighting/switch.py:252]: Shouldn't there be another check where `switches` is empty where the only switch containing a light is off?

Comment (basnijholt) [reply custom_components/adaptive_lighting/switch.py:252]: I did something very similar now. Thanks for the suggestion!

**Comment by basnijholt** [comment]
> It seems odd to have two separate functions defined when all they do is call the logger and raise a valueerror, other than the reviews I left this looks good.

I refactored it again and made it more concise.

File: custom_components/adaptive_lighting/switch.py (hunk 1 above)
Comment (basnijholt) [reply custom_components/adaptive_lighting/switch.py:254]: I added another comment.

**Comment by basnijholt** [comment]
@th3w1zard1 this will merge when you approve :)

**Review by th3w1zard1 — APPROVED** [review]
Clean
